head_test.go

```diff
@@ -295,92 +296,172 @@ func TestHeadDeleteSeriesWithoutSamples(t *testing.T) {
 func TestHeadDeleteSimple(t *testing.T) {
 	numSamples := int64(10)
 
-	head, err := NewHead(nil, nil, nil, 1000)
-	testutil.Ok(t, err)
-	defer head.Close()
-
-	app := head.Appender()
-
-	smpls := make([]float64, numSamples)
-	for i := int64(0); i < numSamples; i++ {
-		smpls[i] = rand.Float64()
-		app.Add(labels.Labels{{"a", "b"}}, i, smpls[i])
-	}
-
-	testutil.Ok(t, app.Commit())
 	cases := []struct {
-		intervals Intervals
-		remaint   []int64
+		intervals     Intervals
+		remaint       []int64
+		remainSampbuf []int64 // Sample buffer that should remain after deletion.
 	}{
 		{
-			intervals: Intervals{{0, 3}},
-			remaint:   []int64{4, 5, 6, 7, 8, 9},
+			intervals:     Intervals{{0, 3}},
+			remaint:       []int64{4, 5, 6, 7, 8, 9},
```

Comment:
I think the current test is more understandable than adding a helper function and passing samples in the test cases, because currently the expected timestamps are clearly visible in the test cases.